iP/src/duke/Duke.java

```diff
@@ -0,0 +1,172 @@
+package duke;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * Duke is the main class in this todo app.
+ */
+public class Duke {
+    private Storage storage;
+    public TaskList tasks;
+    private Ui ui;
+
+    /**
+     * constructor of Duke
+     * @param filePath path where saved data of todo is stored
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            // need empty tasks to load properly
```

Comment:
From the java coding standards: "Logical units within a block should be separated by one blank line."
Consider leaving a line before the try block?